### PR TITLE
Update timeout, tower-test and reconnect

### PR DESCRIPTION
**Pull request by LucioFranco**
`tower-test` needs a better refactor to follow what we do in `tokio/tests`.

### CODE REVIEW

File: tower-test/Cargo.toml [hunk 1]
```diff
@@ -22,9 +22,8 @@ categories = ["asynchronous", "network-programming"]
...
 [dependencies]
-futures-util-preview = "=0.3.0-alpha.19"
-futures-executor-preview = "=0.3.0-alpha.19"
-tokio-test = "=0.2.0-alpha.6"
-tokio-sync = "=0.2.0-alpha.6"
-tower-service = { version = "=0.3.0-alpha.2", path = "../tower-service" }
+futures-util = "0.3"
+tokio = { version = "0.2", features = ["sync"]}
+tokio-test = "0.2"
+tower-service = { version = "0.3", path = "../tower-service" }
```
Comment (jonhoo) [inline tower-test/Cargo.toml]: I thought the idea was to only have `path` dependencies if you depend on unreleased changes?

File: tower-timeout/Cargo.toml [hunk 2]
```diff
@@ -7,22 +7,22 @@ name = "tower-timeout"
...
-version = "0.3.0-alpha.2"
+# - Create "v0.3.x" git tag.
+version = "0.3.0"
 authors = ["Tower Maintainers <team@tower-rs.com>"]
 license = "MIT"
 readme = "README.md"
 repository = "https://github.com/tower-rs/tower"
 homepage = "https://github.com/tower-rs/tower"
-documentation = "https://docs.rs/tower-timeout/0.3.0-alpha.2"
+documentation = "https://docs.rs/tower-timeout/0.3.0"
```
Comment (jonhoo) [inline tower-timeout/Cargo.toml]: We shouldn't need to set these any more. `docs.rs` already takes care of that.

**Comment by jonhoo** [comment]
I wonder if `tower` should also go the route of `tokio` and bundle all these subcrates into one crate with features?

File: tower-test/Cargo.toml (hunk 1 above)
Comment (LucioFranco) [reply tower-test/Cargo.toml]: Not sure it works for now, if we run into issues we can remove 🤷‍♂

Comment (jonhoo) [reply tower-test/Cargo.toml]: I think it causes some pain with people depending on git versions of the `tower-` crates, but I forget exactly how.

**Comment by LucioFranco** [comment]
I am gonna merge this so follow up work can happen.

**Pull request merged**